lib/Controller/PollController.php

```diff
@@ -69,18 +72,22 @@ public function __construct(
 	#[RequireParticipant]
 	#[RequirePermission(permission: RequirePermission::CHAT)]
 	#[RequireReadWriteConversation]
-	public function createPoll(string $question, array $options, int $resultMode, int $maxVotes): DataResponse {
+	public function createPoll(string $question, array $options, int $resultMode, int $maxVotes, bool $draft = false): DataResponse {
 		if ($this->room->isFederatedConversation()) {
 			/** @var \OCA\Talk\Federation\Proxy\TalkV1\Controller\PollController $proxy */
 			$proxy = \OCP\Server::get(\OCA\Talk\Federation\Proxy\TalkV1\Controller\PollController::class);
-			return $proxy->createPoll($this->room, $this->participant, $question, $options, $resultMode, $maxVotes);
+			return $proxy->createPoll($this->room, $this->participant, $question, $options, $resultMode, $maxVotes, $draft);
 		}
 
 		if ($this->room->getType() !== Room::TYPE_GROUP
 			&& $this->room->getType() !== Room::TYPE_PUBLIC) {
 			return new DataResponse([], Http::STATUS_BAD_REQUEST);
 		}
 
+		if ($draft === true && !$this->participant->hasModeratorPermissions()) {
+			return new DataResponse([], Http::STATUS_BAD_REQUEST);
```

Comment:
Should be 403?

Reply:
Well can argue in both ways. The request is bad because you send a prop you can not. Or we send forbidden. The question is what is better/easier to handle as an api consumer.
I'd say we stay with bad request and add the `['error' => 'draft']` similarly to other places

Reply:
Ok. I forgot to hide the option for non-moderators on frontend, when we found out, so shouldn't be a problem